src/datarobotx/idp/retraining_policies.py

```
@@ -0,0 +1,98 @@
+    if dataset_id is not None:
+        deployment = dr.Deployment.get(deployment_id=deployment_id)  # type: ignore
+        prediction_env_id = deployment.default_prediction_server["id"]  # type: ignore
+        get_payload = {"datasetId": dataset_id, "predictionEnvironmentId": prediction_env_id}
+        client.request(
+            method="PATCH", url=f"deployments/{deployment_id}/retrainingSettings", json=get_payload
+        )
+
+    get_response = client.request(
+        method="GET", url=f"deployments/{deployment_id}/retrainingPolicies"
+    )
+    retraining_policies_payload: Dict[str, Any] = json.loads(get_response.text)
+    policies = retraining_policies_payload["data"]
+
+    policy_payload_update = {"name": name, **kwargs}
+    response: Any = None
+    retraining_policy_id: str = ""
+
+    for policy in policies:
+        if policy["name"] == name:
+            retraining_policy_id = policy["id"]
+            response = client.request(
+                method="PATCH",
+                url=f"deployments/{deployment_id}/retrainingPolicies/{retraining_policy_id}",
+                json=policy_payload_update,
+            )
+            break  # No need to continue once the policy is found and updated
+
+    if response is None:
+        response = client.request(
+            method="POST",
+            url=f"deployments/{deployment_id}/retrainingPolicies",
+            json=policy_payload_update,
+        )
+
+    if retraining_policy_id == "":
+        get_response_id = client.request(
+            method="GET", url=f"deployments/{deployment_id}/retrainingPolicies"
+        )
+        payload: Dict[str, Any] = json.loads(get_response_id.text)
+        policies = payload["data"]
+        for policy in policies:
+            if policy["name"] == name:
+                retraining_policy_id = policy["id"]
+                break
+
+    return retraining_policy_id if retraining_policy_id != "" else "Policy creation failed."
```

Should raise an error rather than just return a fail string